libs/pdf-explorer/src/lib/pdf-explorer.ts

```diff
@@ -0,0 +1,46 @@
+import fetch from 'node-fetch';
+import pdf from 'pdf-parse';
+import { constants } from './pdf-explorer-constants';
+
+/**
+ * Check whether the paragraphs contains valid information
+ */
+function paragraphContainsInformation(body: string): boolean {
+  return body.length > constants.MIN_CHAR_COUNT_PER_PARAGRAPH && body.includes('.');
+}
+
+function getCharCount(text: string, pattern: RegExp): number {
+  return (text.match(pattern) || []).length;
+}
+
+function groupSentencesIntoParagraphs(sentences: string[]): string[] {
+  let syntheticParagraphsInd = 0;
+  const syntheticParagraphs: string[] = [];
+  sentences.forEach((sentence, i) => {
+    if (!syntheticParagraphs[syntheticParagraphsInd]) {
+      syntheticParagraphs[syntheticParagraphsInd] = sentence + '. ';
+    } else {
+      syntheticParagraphs[syntheticParagraphsInd] += sentence + '. ';
+    }
+    if (i % constants.SENTENCES_PER_PARAGRAPH === 0) {
+      syntheticParagraphsInd++;
+    }
+  });
+  return syntheticParagraphs;
+}
+
+export function splitTextToParagraphs(rawText: string): string[] {
+  const tabCount = getCharCount(rawText, /\t/g);
+  if (tabCount > constants.MIN_TAB_COUNT) {
+    return rawText.split('\t').filter(paragraphContainsInformation);
+  }
+  const sentences = rawText.split('.');
```

Comment:
Instead of having to try to group sentences into a paragraph, could you split by either \t OR \n?

Reply:
Ok sure, it tries by \t initially. I will also then add \n. It falls back to by sentences to make sure that paragraphs are not too short

Reply:
I gave \n a test. Each new line in the pdf is then split and considered a paragraph. So, any line break would cause a distinction. I added a comment explaining the function as well